src/main/kotlin/org/rust/lang/core/psi/ext/RsFunction.kt

```diff
@@ -135,7 +135,10 @@ val RsFunction.isCustomDeriveProcMacroDef: Boolean
     get() = queryAttributes.hasAttribute("proc_macro_derive")
 
 val RsFunction.isProcMacroDef: Boolean
-    get() = queryAttributes.hasAnyOfOuterAttributes(
+    get() = greenStub?.isProcMacroDef ?: (queryAttributes.isProcMacroDef)
+
+val QueryAttributes.isProcMacroDef
```

Comment:
Do we really need this as separate public property?
Maybe it's worth to unify it with other similar properties and inline it into `RsFunction.isProcMacroDef`?

Reply:
This and similar properties specially separated from RsFunction because QueryAttributes is constructed differently during stub building

Reply:
This and similar properties specially separated from RsFunction because QueryAttributes is constructed differently during stub building